pkg/controller/projectclaim/projectclaimadapter.go

```diff
@@ -363,29 +275,28 @@ func (c *ProjectClaimAdapter) EnsureProjectReferenceExists() (gcputil.OperationR
 	return gcputil.ContinueProcessing()
 }
 
+// Sets a ProjectClaim State to Pending for the case ProjectReference CR is being set up
 func (c *ProjectClaimAdapter) EnsureProjectClaimStatePending() (gcputil.OperationResult, error) {
 	return c.EnsureProjectClaimState(gcpv1alpha1.ClaimStatusPending)
 }
 
+// Sets a ProjectClaim State to PendingProject for the case gcp project for the ProjectReference is being prepared
 func (c *ProjectClaimAdapter) EnsureProjectClaimStatePendingProject() (gcputil.OperationResult, error) {
 	return c.EnsureProjectClaimState(gcpv1alpha1.ClaimStatusPendingProject)
 }
 
+// Sets the Status of a ProjectClaim
 func (c *ProjectClaimAdapter) EnsureProjectClaimState(state gcpv1alpha1.ClaimStatus) (gcputil.OperationResult, error) {
 	if c.projectClaim.Status.State == state {
 		return gcputil.ContinueProcessing()
 	}
 
-	if state == gcpv1alpha1.ClaimStatusPending && c.projectClaim.Status.State != gcpv1alpha1.ClaimStatusError {
-		if c.projectClaim.Status.State != "" {
```

Comment:
Good point, this part is not covered by any unit test and the condition was not very understandable to me.
I thought this could be simplified, but this really changes the functionality. Unfortunately, this is not covered by any unit test, I will revert my changes.
I believe we are basically ensuring that the we only have ProjectClaim state transitions from:
**"" -> Pending
error -> Pending**